src/App/App.js

```diff
@@ -28,8 +29,34 @@ class App extends Component {
           userDeckNames.push(deck.deckName)
         }
       })
-    })
-    this.setState({ userDecks: newUserDecks });
+    });
+    this.setDeckPrices(newUserDecks);
+  }
+
+  setDeckPrices(userDecks) {
+    const newUserDecks = userDecks.map(deck => {
+      deck.price = 0;
+      Object.keys(deck.cardCounts).forEach(cardInDeck => {
+        const cardPrice = this.state.cards.find(card => {
+          return card.cardName === cardInDeck;
+        }).price;
+        const userCard = this.state.userCardsData.find(card => {
+          return card.cardName === cardInDeck
+        });
+        if (userCard) {
+          const priceMultiplier = Math.max(0, (deck.cardCounts[cardInDeck] - userCard.cardCount))
+          deck.price += cardPrice * priceMultiplier;
+        } else {
+          deck.price += cardPrice * deck.cardCounts[cardInDeck];
+        }
+      });
+      deck.price = deck.price.toFixed(2);
+      return deck;
+    });
+    this.setState({
+      userDecks: newUserDecks,
+      cardAreaView: 'compareDecks'
+    });
```

Comment:
This method is really long, but I'm not sure if it can be shortened.

The method maps over all the decks that the user can build with the cards they have. On each iteration, all the cards in the deck are iterated over and the card prices are found.

If the user has that card, the card's price is multiplied by the difference of the card count required by the deck and the card count of the user. The multiplier can be a minimum of 0, so if the deck calls for 3 of a certain card, and the user has 4, the product of the price and the multiplier will not be negative.

This product of price and multiplier is then added to the total price of the deck, and after this is done for every deck, the decks with their new price property are set in state.

I did not write a test for this method because on line 40, it requires this.state.cards which is set in state by a fetch call.